packages/report-server/src/reportBuilder/transform/aliases/summaryAliases.ts

```diff
@@ -0,0 +1,61 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import { Row } from '../../types';
+
+/**
+ * Inserts a new column with a value that is the calculation of the percentage of 1s out of 1s and 0s in a row.
+ * { rowFieldName: 'Availability of Male Condoms', facilityNameA: 'N', facilityNameB: 'N', facilityNameC: 'Y', facilityNameD: 'N'  }
+ *  => { rowFieldName: 'Availability of Male Condoms', facilityNameA: 'N', facilityNameB: 'N', facilityNameC: 'Y', facilityNameD: 'N', rowSummary: 0.75 }
+ */
+
+const detectColumnsToSummarise = (rows: Row[]): string[] => {
+  console.log(rows)
+  const anyRelevantColumns = rows.reduce((previousValue, currentValue) => {
+    const columnsWithYorN = Object.entries(currentValue).filter(([,value]) => value === 'Y' || value === 'N').map(([column,]) => column);
+    return previousValue.concat(columnsWithYorN);
+  },[] as string[])
+  const columnsToSummarise = anyRelevantColumns.reduce((previousValue: string[], currentValue: string): string[] => {
+    if (previousValue.indexOf(currentValue) > -1) {
+      return previousValue
+    }
+    previousValue.push(currentValue)
+    return previousValue
+  },[] as string[])
+
+  return columnsToSummarise
+};
```

Comment:
One slight concern with this function, I think it would classify 'any columns that have a `Y` or an `N` in them' as a column to summarise? If that how we want it to behave then that's okay, but perhaps it's more correct for it to be 'any columns that have only `Y`s or `N`s in them'?

Reply:
@rohan-bes how would we treat undefined. If we looped through each row and some columns were undefined, we would not want to exclude them. It's really only columns with anything other than 'Y' 'N' or undefined that we would exclude.

Reply:
We could do it that way as well 👍 'any column that has only `Y`s, `N`s, or `undefined`'

Reply:
@rohan-bes I've created a filter within detectColumnsToSummarise.